### PR TITLE
List Comprehension example of syntax extension.

### PR DESCRIPTION
**Pull request by lovettchris**
Many thanks to Kyle Miller who first posted this solution on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/List.20Functor), this sample then tests all the [list comprehension features of Python](https://docs.python.org/3/tutorial/datastructures.html) now work in Lean.

### CODE REVIEW

File: ListComprehension/README.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
+# List Comprehension with Syntax Extensions
+
+For those of you who are addicted to the really cool
+[list comprehension features of Python](https://docs.python.org/3/tutorial/datastructures.html),
+you might be looking for something similar in Lean.
+
+First you need to understand that the Lean `List` is a `Functor`.
```
Comment (Kha) [inline ListComprehension/README.md]: That's a somewhat scary sentence considering that many people are familiar with `map` without ever having heard of functors.

**Comment by Kha** [comment]
There is a bit of inconsistent capitalization of "Lean" and "Python"

**Comment by lovettchris** [comment]
fixed, thanks.

File: ListComprehension/README.md (hunk 1 above)
Comment (lovettchris) [reply ListComprehension/README.md]: I've softened the language a bit, I still want to mention functors, but I removed the "need to understand" part.

**Pull request merged**